std/typecons.d

```diff
@@ -5205,34 +5217,76 @@ unittest
         int x;
         this()     {x = 0;}
         this(int i){x = i;}
+        ~this()    {}
     }
 
-    // Standard usage
+    // Standard usage, constructing A on the stack
     auto a1 = scoped!A();
-    auto a2 = scoped!A(1);
     a1.x = 42;
-    assert(a1.x == 42);
-    assert(a2.x ==  1);
+    
+    // Result of `scoped` call implicitly converts to a class reference
+    A aRef = a1;
+    assert(aRef.x == 42);
+    
+    // Scoped destruction
+    {
+        auto a2 = scoped!A(1);
+        assert(a2.x == 1);
+        aRef = a2;
+        // a2 is destroyed here, calling A's destructor
+    }
+    // aRef is now an invalid reference
+
+    // Here the temporary scoped A is immediately destroyed.
+    // This means the reference is then invalid.
+    version(InvalidReference)
+    {
+        // Wrong, should use `auto`
+        A invalid = scoped!A();
+    }
 
+    // Use with alias
+    alias makeScopedA = scoped!A;
+    auto a3 = makeScopedA();
+    auto a4 = makeScopedA(1);
+    
     // Restrictions
     static assert(!is(typeof({
```

Comment:
I think `move` probably always did work with `scoped`. In my updated https://github.com/D-Programming-Language/phobos/pull/4129, I changed the assert to `version(Bug)` instead.